src/main/webapp/step3-startup.js

```diff
@@ -1,36 +1,95 @@
 import Step3Renderer from './renderers/Step3Renderer.js';
+import MapsInterface from './utilities/MapsInterface.js';
 
 // TODO(tzavidas): replace this array with the event management class, as discussed with remusn
 const events = [{
         name: 'Breakfast',
-        location: 'Zurich',
-        startingTime: '9:30',
-        endingTime: '10:30',
+        location: 'Google Zurich',
+        startingTime: new Date('2020-09-22T09:30:00'),
+        endingTime: new Date('2020-09-22T10:00:00'),
+        travelMode: 'TRANSIT',
     }, {
         name: 'Lunch',
-        location: 'Stuttgart',
-        startingTime: '13:30',
-        endingTime: '14:30',
+        location: 'Zurich Main Station',
+        startingTime: new Date('2020-09-22T13:30:00'),
+        endingTime: new Date('2020-09-22T14:30:00'),
+        travelMode: 'TRANSIT',
     }, {
         name: 'Dinner',
-        location: 'Bucharest',
-        startingTime: '19:30',
-        endingTime: '20:30',
+        location: 'Bar am Wasser',
+        startingTime: new Date('2020-09-22T19:30:00'),
+        endingTime: new Date('2020-09-22T20:30:00'),
     }
 ];
 
-/**
- * Empties the event list
- * Add the events from local storage
- */
+// For each event, it gets the location, converts it into coordinates and adds it to the object
+async function getLocationsInCoordinates(eventsList, mapsInterface) {
+    const locationsConverted = [];
+    let hasErrors = false;
+
+    for(const event of eventsList) {
+        const { status: reqStatus, coordinates } = await mapsInterface.convertToCoordinates(event.location);
+
+        if(reqStatus === 'OK') {
+            event.coordinates = coordinates;
+        } else {
+            event.coordinates = null;
```

Comment:
I'm not sure if omitting event is a good idea, resoult route will lead from place A to C directly (if B omitted) and that is confusing, but I don't have a better idea...

Reply:
Is it okay if we leave it like this now? The idea is that the location names will be validated on step1 (we can query the geolocation API and check if it's valid). So, it will be guaranteed that, on step3, everything will work fine!

Reply:
As we don't have better idea - keep it as is.

Reply:
Added a TODO!